tests/Struct/FindologicProductTest.php

```diff
@@ -189,6 +188,86 @@ public function testNoProductCategories(): void
         );
     }
 
+    public function parentAndChildrenCategoryProvider(): array
+    {
+        return [
+            'Parent and children have the same categories assigned' => [
+                'isParentAssigned' => true,
+                'isVariantAssigned' => true,
+            ],
+            'Parent has no categories and children have some categories assigned' => [
+                'isParentAssigned' => false,
+                'isVariantAssigned' => true
+            ],
+            'Parent has categories and children have no categories assigned' => [
+                'isParentAssigned' => true,
+                'isVariantAssigned' => false
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider parentAndChildrenCategoryProvider
+     */
+    public function testOnlyUniqueCategoriesAreExported(bool $isParentAssigned, bool $isVariantAssigned): void
```

Comment:
Where are the test cases for

* Create a product with 5 variants. Only assign a category to one of the variants and ensure it is exported properly.
* Remove the set category of the previously created variant. Calling the export now should result in the product being skipped. Ensure that the logs contain information that a ProductHasNoCategoriesException is thrown.

Reply:
Already replied on Youtrack for this. This is a click test scenario. The unit tests cover the cases.